feg/gateway/services/envoy_controller/servicers/envoy_controller.go

```diff
@@ -35,10 +35,36 @@ func (s *envoyControllerService) AddUEHeaderEnrichment(
 ) (*protos.AddUEHeaderEnrichmentResult, error) {
 	glog.Infof("AddUEHeaderEnrichmentResult received for IP %s", req.UeIp.Address)
 	glog.Infof("req %s", (req))
-	s.ue_infos = add_to_list(s.ue_infos, req)
-	s.controller_cli.UpdateSnapshot(s.ue_infos)
 
-	return &protos.AddUEHeaderEnrichmentResult{}, nil
+	ueIp := string(req.UeIp.Address)
+
+	if _, ok := s.ueInfos[ueIp]; !ok {
+		s.ueInfos[ueIp] = map[string]*control_plane.UEInfo{}
+	} else {
+		if _, ok := s.ueInfos[ueIp][req.RuleId]; ok {
+			return &protos.AddUEHeaderEnrichmentResult{Result: protos.AddUEHeaderEnrichmentResult_RULE_ID_CONFLICT}, nil
+		}
+	}
+
+	// Loop over other rules, check if there is a conflict with the Websites (2 identical Websites will cause an envoy deadloop)
```

Comment:
no need to change it, but in case you wanted to avoid so much loops you could put `ueINFO` and `Website` in a set (go doesnt have set, so you use a map). Make sure they are hasheable 

```
existing_set_ue_infos := make(map[ue_info_type]bool) 
for _, ue_info := range s.ueInfos[ueIp] {
	set_ue_infos[ue_info]= true 
}
existing_set_websites := make(map[websites_type]bool) 
for _, ue_info := range s.ueInfos[ueIp] {
	existing_set_websites[ue_info]= true 
}


```